sdk/core/azure-core/inc/azure/core/internal/json/json_optional.hpp

```diff
@@ -37,11 +37,24 @@ namespace Azure { namespace Core { namespace Json { namespace _internal {
     static inline void SetIfExists(
         Azure::Nullable<T>& destination,
         Azure::Core::Json::_internal::json const& jsonKey,
-        std::string const& key) noexcept
+        std::string const& key)
     {
       if (jsonKey.contains(key) && !jsonKey[key].is_null()) // In Json and not-Null
       {
-        destination = jsonKey[key].get<T>();
+        try
+        {
+          destination = jsonKey[key].get<T>();
+        }
+        catch (std::exception const& e)
```

Comment:
I didn't want to catch the specific `type_error` for two reasons:
1) This is a custom error in the `detail` namespace from nlohmann, which might imply it could change.
2) Though the `.get<T>()` call is likely never going to fail with other exception types, I wanted to make sure we catch any exceptions that nlohmann throws.

I am open to other thoughts on this.